src/luaharfbuzz/buffer.c

```diff
@@ -96,8 +96,10 @@ static int buffer_add_utf8(lua_State *L) {
   Buffer *b = (Buffer *)luaL_checkudata(L, 1, "harfbuzz.Buffer");
 
   const char *text = luaL_checkstring(L, 2);
+  unsigned int item_offset = luaL_checkint(L,3);
+  int item_length = luaL_checkint(L,4);
 
-  hb_buffer_add_utf8(*b, text, strlen(text), 0, strlen(text));
+  hb_buffer_add_utf8(*b, text, strlen(text), item_offset, item_length);
```

Comment:
For consistency you might want to use `-1` instead of `strlen(text)` here as well, but is probably makes no real difference.